### ISSUE TITLE
enable-perms causes geesefs to crash

### ISSUE THREAD

**Issue by mingfang** [yandex-cloud/geesefs#66]
Here is the crash log

```
csi-s3 attributes map[bucket:rebelsoft-s3fs capacity:1073741824 mounter:geesefs options:--memory-limit 1000 --enable-perms storage.kubernetes.io/csiProvisionerIdentity:1677274281711-8081-ru.yandex.s3.csi]      
csi-s3 mountflags []                                                                                                                                                                                              
csi-s3 I0227 16:59:24.008291       1 mounter.go:63] Mounting fuse with command: geesefs and args: [--endpoint https://s3.us-east-1.amazonaws.com -o allow_other --log-file /dev/stderr --memory-limit 1000 --enabl
e-perms rebelsoft-s3fs:pvc-9798eb15-8a30-4d5c-bba2-8d57e6d953a0 /var/lib/kubelet/pods/a370f5b5-5dd2-41aa-99c8-1d8fa7294422/volumes/kubernetes.io~csi/pvc-9798eb15-8a30-4d5c-bba2-8d57e6d953a0/mount]              
csi-s3 2023/02/27 16:59:24.204538 main.INFO File system has been successfully mounted.                                                                                                                            
csi-s3 I0227 16:59:25.206252       1 nodeserver.go:117] s3: volume rebelsoft-s3fs/pvc-9798eb15-8a30-4d5c-bba2-8d57e6d953a0 successfully mounted to /var/lib/kubelet/pods/a370f5b5-5dd2-41aa-99c8-1d8fa7294422/volu
mes/kubernetes.io~csi/pvc-9798eb15-8a30-4d5c-bba2-8d57e6d953a0/mount                                                                                                                                              
csi-s3 panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                    
csi-s3 [signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0xe7f809]                                                                                                                                    
csi-s3                                                                                                                                                                                                            
csi-s3 goroutine 101 [running]:                                                                                                                                                                                   
csi-s3 github.com/yandex-cloud/geesefs/internal.(*Inode).cloud(0x0, 0x300000001000000, 0x0, 0xc00093fde8, 0xc00093fde8)                                                                                           
csi-s3     /home/runner/work/geesefs/geesefs/internal/handles.go:248 +0x29                                                                                                                                        
csi-s3 github.com/yandex-cloud/geesefs/internal.(*Inode).SendMkDir(0xc0001768c0)                                                                                                                                  
csi-s3     /home/runner/work/geesefs/geesefs/internal/dir.go:1223 +0x47                                                                                                                                           
csi-s3 github.com/yandex-cloud/geesefs/internal.(*Inode).TryFlush(0xc0001768c0, 0xc000149200)                                                                                                                     
csi-s3     /home/runner/work/geesefs/geesefs/internal/file.go:1274 +0x1d8                                                                                                                                         
csi-s3 github.com/yandex-cloud/geesefs/internal.(*Goofys).Flusher(0xc000130000)                                                                                                                                   
csi-s3     /home/runner/work/geesefs/geesefs/internal/goofys.go:608 +0xf7                                                                                                                                         
csi-s3 created by github.com/yandex-cloud/geesefs/internal.newGoofys                                                                                                                                              
csi-s3     /home/runner/work/geesefs/geesefs/internal/goofys.go:282 +0xad1   
```

**Comment by vitalif** [comment]
Hi, it doesn't reproduce so simply for me. Can you give me a minimal example? I tried to mount an empty bucket in AWS with --enable-perms and create a directory but didn't get this crash.

By the way, permissions won't work correctly with AWS - see README:
- File mode/owner/group, symbolic links, custom mtimes and special files (block/character devices, named pipes, UNIX sockets) are supported, but they are restored correctly only when using Yandex S3 because standard S3 doesn't return user metadata in listings and reading all this metadata in standard S3 would require an additional HEAD request for every file in listing which would make listings too slow.

But it shouldn't crash of course.

**Comment by vitalif** [comment]
I got a bug report for (probably) the same problem and fixed it in 0.34.5: https://github.com/yandex-cloud/geesefs/releases/tag/v0.34.5
Presumably it wasn't related to --enable-perms, it rather was a race condition between rename and listing.
Please update and recheck. Reopen the bug if it still persists.
